compiler/llvm/clangUtil.cpp

```diff
@@ -4444,13 +4444,38 @@ static void moveResultFromTmp(const char* resultName, const char* tmpbinname);
 static llvm::CodeGenFileType getCodeGenFileType() {
   switch (getGpuCodegenType()) {
     case GpuCodegenType::GPU_CG_AMD_HIP:
-      return llvm::CodeGenFileType::CGFT_ObjectFile;
```

Comment:
I'm game with leaving this code in, but could you add a comment explaining why it exists?

Reply:
Sure, I've added some comments here: https://github.com/chapel-lang/chapel/pull/22862/commits/368e798b2eae6d4bb77a30fe1f2a433283e630ad